$OpenBSD: patch-src_PrefsUnifiedDlg_cpp,v 1.1 2011/12/11 18:31:13 dcoppa Exp $

Fix "Prompt on exit" preference
(upstream git commit aea8d002b6446dcaf38db4d4ab222fcebce60948)

Fix: require restart when protocol obfuscation setting gets changed
(upstream git commit ec2e66216738f92724a37fa030a79734e0e8b1ba)

--- src/PrefsUnifiedDlg.cpp.orig	Sun Oct  9 02:19:08 2011
+++ src/PrefsUnifiedDlg.cpp	Fri Dec  9 16:35:56 2011
@@ -252,6 +252,7 @@ wxDialog(parent, -1, _("Preferences"),
 				FindWindow(IDC_MINTRAY)->Show(false);
 			#else
 				FindWindow(IDC_MACHIDEONCLOSE)->Show(false);
+				thePrefs::SetHideOnClose(false);
 			#endif
 		} else if (pages[i].m_function == PreferencesEventsTab) {
 
@@ -438,7 +439,7 @@ bool PrefsUnifiedDlg::TransferToWindow()
 
 	FindWindow(IDC_MACHIDEONCLOSE)->Enable(true);
 	FindWindow(IDC_EXIT)->Enable(!thePrefs::HideOnClose());
-	if (!thePrefs::HideOnClose()) {
+	if (thePrefs::HideOnClose()) {
 		CastChild(IDC_EXIT, wxCheckBox)->SetValue(false);
 	}
 
@@ -612,6 +613,10 @@ void PrefsUnifiedDlg::OnOk(wxCommandEvent& WXUNUSED(ev
 	if (CfgChanged(IDC_EXT_CONN_IP)) {
 		restart_needed = true;
 		restart_needed_msg += _("- External connect interface changed.\n");
+	}
+	if (CfgChanged(IDC_SUPPORT_PO)) {
+		restart_needed = true;
+		restart_needed_msg += _("- Protocol obfuscation support changed.\n");
 	}
 
 	// Force port checking
